lib/Common/CommonBasic.h

```diff
@@ -8,6 +8,26 @@
 #include "CommonDefines.h"
 #define _CRT_RAND_S         // Enable rand_s in the CRT
 
+// AddressSanitizer: check if an address is in asan fake stack
+#ifdef __has_feature
```

Comment:
nit: we also have 

```
#define NO_SANITIZE_ADDRESS
#if defined(__has_feature)
#if __has_feature(address_sanitizer)
```

under CommonDefines.h .. It could be nice to define `__has_feature` if it wasn't defined

```
#ifndef __has_feature
#define __has_feature(has) 0
#endif
```

under CommonDefines.h and not repeating `#ifdef __has_feature` on different places.

Maybe also combine two `address_sanitizer` related code here or there?

Reply:
Thought about the same, but hesitated to put these extra code (extern declarations, inline function) in CommonDefines.h which mostly contains short defines. NO_SANITIZE_ADDRESS seems more broadly applicable and CommonDefines.h is directly included by some header files, so it is also risky to move NO_SANITIZE_ADDRESS definition out. I'll try simplify with __has_feature definition.